#!/bin/Rscript
## DEG analysis using DESeq2
## Usage: Rscript DESeq2.R count_matrix_file design_table outfile
##
## 		  count_matrix: generated by featureCounts
##		  design_table: including sample infomations, especially with its conditions
## 		  outfile: Gene differential expression table with p-values
##

library(DESeq2)

# Define a function for DEG analysis
DEG_analysis <- function(counts_file, design_table, result_file) {
	# Load counts generated by featureCounts
	cts <- read.delim(counts_file, skip=1) 
	rownames(cts) <- cts$Geneid
	cts <- as.matrix(cts[,-c(1:6)]) # remove not-used information
	samples <- sub(".Aligned.out.sam", "", colnames(cts))
	colnames(cts) <- samples
	
	# Load sample information table: coldata
	coldata <- read.delim(design_table)
	rownames(coldata) <- coldata[,1]
	coldata <- coldata[,-1]
	# make the two table consistent: rownames(coldata) == colnames(cts)
	cts <- cts[,rownames(coldata)]

	# Pre-filtering: keep only rows that have at least 10 reads total
	keep <- rowSums(cts) >= 10
	cts <- cts[keep,]
	
	# Generate DESeq2 dataset
	dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ condition)
	#dds
	# DEG analysis
	dds <- DESeq(dds)
	resultsNames(dds) # lists the coefficients
	res <- results(dds)
	
	# Filter rows with NA padj
	keep <- !is.na(res$padj)
	res <- res[keep,]
	# output results
	res <- data.frame(Geneid=rownames(res), res)
	write.table(res, file=result_file, quote=F, sep='\t', row.names=F)
}

# Load parameters
args <- commandArgs(trailingOnly=TRUE)
counts_file <- args[1]
design_table <- args[2]
outfile <- args[3]

DEG_analysis(counts_file, design_table, outfile)

